### ISSUE TITLE
Grouped by issue with type conversion

### ISSUE THREAD

**Issue by pdeffebach** [JuliaData/DataFramesMeta.jl#98]
```
using DataFrames, DataFramesMeta
df = DataFrame(
    id = [1,1,1,2,2,2,3,3,3],
    a = ["foo", "bar", "baz","foo", "bar", "baz","foo", "bar", "baz"],
    b = ["x", missing, "z","y", missing, missing,"x", missing, "z"],
    c = [missing, "w", missing, missing, "w", missing, missing,"w", missing],
    d = [missing, missing, 10, missing, missing, 20, missing, missing, 15],
    c_number = [missing, 1, missing, missing, 1, missing, missing, 1, missing],
    )

julia> df2 = @linq df |>
       groupby(:id) |>
       @transform(newc = maximum(skipmissing(:c)))
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Char
This may have arisen from a call to the constructor Char(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] setindex!(::Array{Char,1}, ::String, ::UnitRange{Int64}) at .\array.jl:591
 [2] insert_multiple_entries!(::DataFrames.DataFrame, ::String, ::UnitRange{Int64}, ::Symbol) at C:\Users\Pdeffebach\.julia\v0.6\DataFrames\src\dataframe\dataframe.jl:373
 [3] setindex!(::DataFrames.DataFrame, ::String, ::UnitRange{Int64}, ::Symbol) at C:\Users\Pdeffebach\.julia\v0.6\DataFrames\src\dataframe\dataframe.jl:511
 [4] #transform#40(::Array{Any,1}, ::Function, ::DataFrames.GroupedDataFrame) at C:\Users\Pdeffebach\.julia\v0.6\DataFramesMeta\src\DataFramesMeta.jl:400
 [5] (::DataFramesMeta.#kw##transform)(::Array{Any,1}, ::DataFramesMeta.#transform, ::DataFrames.GroupedDataFrame) at .\<missing>:0

julia> df2 = @linq df |>
       groupby(:id) |>
       @transform(newc = maximum(skipmissing(:c_number)))
9×7 DataFrames.DataFrame
│ Row │ id │ a   │ b       │ c       │ d       │ c_number │ newc │
├─────┼────┼─────┼─────────┼─────────┼─────────┼──────────┼──────┤
│ 1   │ 1  │ foo │ x       │ missing │ missing │ missing  │ 1    │
│ 2   │ 1  │ bar │ missing │ w       │ missing │ 1        │ 1    │
│ 3   │ 1  │ baz │ z       │ missing │ 10      │ missing  │ 1    │
│ 4   │ 2  │ foo │ y       │ missing │ missing │ missing  │ 1    │
│ 5   │ 2  │ bar │ missing │ w       │ missing │ 1        │ 1    │
│ 6   │ 2  │ baz │ missing │ missing │ 20      │ missing  │ 1    │
│ 7   │ 3  │ foo │ x       │ missing │ missing │ missing  │ 1    │
│ 8   │ 3  │ bar │ missing │ w       │ missing │ 1        │ 1    │
│ 9   │ 3  │ baz │ z       │ missing │ 15      │ missing  │ 1    │
```

Why is this operation trying to convert the column `:c` to a column of `Char` in this operation? Weirdly, the non-spreading version, with `@by` works fine. 

```
julia> df2 = @linq df |>
       @by(:id, newc = maximum(skipmissing(:c)))
3×2 DataFrames.DataFrame
│ Row │ id │ newc │
├─────┼────┼──────┤
│ 1   │ 1  │ w    │
│ 2   │ 2  │ w    │
│ 3   │ 3  │ w    │


```

Can someone replicate this issue on 0.7? I understand there are changes to the way conversions and constructors work. But this is an odd error to happen. let me know if you have any idea for what code is affecting this and I could work on a PR.

**Comment by pdeffebach** [comment]
I just tested some of this on 0.7. 

It looks like `transform(x  = 2)` works fine, which I guess is unsurprising, given the way `DataFramesMeta` works. I'm not sure how to get dot assignment to work according to 0.7 rules.

**Comment by pdeffebach** [comment]
fixed with #101
